ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/GSM.hs

```diff
@@ -125,77 +125,86 @@ sm ::
      IOLike m
   => Maybe UpstreamPeer
   -> Vars m
-  -> LedgerStateJudgement
+  -> Context
   -> QSM.StateMachine Model Command m Response
-sm ub vars j = QSM.StateMachine {
+sm ub vars ctx = QSM.StateMachine {
     QSM.cleanup = \_model -> pure ()
   ,
     QSM.generator = generator ub
   ,
-    QSM.initModel = initModel j
+    QSM.initModel = initModel ctx
   ,
     QSM.invariant = Nothing
   ,
     QSM.mock = \model cmd -> pure $ mock model cmd
   ,
     QSM.postcondition = postcondition
   ,
-    QSM.precondition = precondition
+    QSM.precondition = precondition ctx
   ,
     QSM.semantics = semantics vars
   ,
     QSM.shrinker = shrinker
   ,
-    QSM.transition = transition
+    QSM.transition = transition ctx
   }
 
 prop_sequential ::
      Maybe UpstreamPeer
   -> LedgerStateJudgement
+  -> QC.Fun (Set.Set UpstreamPeer) Bool
   -> QC.Property
-prop_sequential ub j0 =
+prop_sequential ub j0 (QC.Fn isHaaSatisfied) =
     QSM.forAllCommands
         commandArbitrarySM
         mbMinimumCommandLen
-        (prop_sequential1 j0)
+        (prop_sequential1 ctx)
   where
+    ctx = Context {
+        cInitialJudgement = j0
+      ,
+        cIsHaaSatisfied = isHaaSatisfied
+      }
+
     mbMinimumCommandLen = Just 20   -- just a guess
 
     -- NB the monad is irrelevant here but is ambiguous, so we merely ascribe a
     -- convenient concrete one
-    commandArbitrarySM = sm ub (undefined :: Vars IO) j0
+    commandArbitrarySM = sm ub (undefined :: Vars IO) ctx
 
 prop_sequential1 ::
-     LedgerStateJudgement
+     Context
   -> QSM.Commands Command Response
   -> QC.Property
-prop_sequential1 j0 cmds = runSimQC $ do
+prop_sequential1 ctx cmds = runSimQC $ do
+    let s0 = case j0 of
```

Comment:
Maybe `s0` and `j0` could have more descriptive names.

Reply:
Renamed :+1: 83cd2fe833a02e3a557129a68b41ae783b2c4e38